[qinxutan] iP

qinxutan opened the pull request:
# YapperPro
> "You only live once." - Yapper

YapperPro is your best friend. It takes down things you have to do that you can't stop yapping about. It's
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/qinxutan/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it's **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines(coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

qinxutan committed:
Rename Duke to Yapper

qinxutan committed:
Echo

qinxutan committed:
Add list

qinxutan committed:
ToDos, Events, Deadlines

qinxutan committed:
UI Testing

qinxutan committed:
Handle errors.

qinxutan committed:
Delete

qinxutan committed:
Change error message

qinxutan committed:
Save tasks in hard disk automatically when task list changes

qinxutan committed:
Teach chatbox to understand dates

qinxutan committed:
Teach chatbox to understand date and time

qinxutan committed:
Make code more OOP

qinxutan committed:
Organize into Packages

qinxutan committed:
Use Gradle to automate testing

qinxutan committed:
Implemented build.gradle part 2

qinxutan committed:
Add Junit Tests

qinxutan committed:
Package the App as a JAR file

qinxutan committed:
Add JavaDoc comments

qinxutan committed:
Follow coding standard

qinxutan committed:
Implement find feature

qinxutan committed:
Check style

qinxutan committed:
Add GUI to yapper

qinxutan committed:
Checkstyle

qinxutan committed:
UI changes

Teeesa7 commented on src/main/java/yapper/Storage.java:
```diff
@@ -0,0 +1,58 @@
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
Is this import statement supposed to start from line 9 instead? I noticed the import statements not being in the right order at different places, so just take note of that  :)

Teeesa7 commented on src/main/java/yapper/Yapper.java:
```diff
@@ -0,0 +1,253 @@
+     */
+    public String showWelcomeMessage() {
+        String message = "Hello! I'm Yapper. \n"
+                + "What would you like to yap about today? :-)";
```
I like your welcome message 🤣

Teeesa7 commented on src/main/java/yapper/command/Parser.java:
```diff
@@ -0,0 +1,60 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import yapper.YapperException;
```
Perhaps you can put this in a new package called "exception"?

Teeesa7 commented on src/main/java/yapper/tasks/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public String getType() {
+        return "D";
```
For the type of tasks, you can consider putting them as an enum instead?

Teeesa7 reviewed (commented):
LGTM! The code follows almost all the quality guidelines.

qinxutan committed:
Subject:
Utilize assert statements for documenting crucial assumptions in code

Body:
Incorporate assert statements throughout the codebase to document significant assumptions that should hold at various points during execution. By leveraging the assert feature (not JUnit assertions), I enhance code readability and maintainability by explicitly stating conditions that must be true for the code to function correctly.

Changes Made:
- Introduced assert statements at critical junctures in the code to verify the validity of important assumptions.
- Ensured that assert statements are used judiciously, focusing on pivotal conditions that are fundamental to the program's logic and behavior.
- Documented each assert statement with a clear and concise message explaining the expected condition.

Rationale:
- Assert statements serve as documentation for developers, outlining essential assumptions and expectations in the code.
- Utilizing assert statements aids in debugging by immediately highlighting any violations of critical conditions during program execution.
- Enhances code reliability by enforcing essential preconditions and invariants, contributing to overall code quality.

Scope/Category:
Code Quality Enhancement

qinxutan committed:
Subject:
Refactor code for improved code quality and maintainability

Body:
Conducted a critical examination of the existing codebase to identify areas for enhancement in terms of code quality and maintainability. Implemented several refactorings to address identified issues and improve overall code quality. The following changes were made:

Changes Made:
- Simplified complex logic by breaking down lengthy functions into smaller, more manageable units, enhancing readability and comprehension.
Introduced meaningful variable names and method names to improve code clarity and convey intent effectively.
- Removed duplicate code segments and consolidated common functionalities to reduce redundancy and promote DRY (Don't Repeat Yourself) principles.
- Employed consistent coding style and formatting conventions throughout the codebase for improved uniformity and readability.
- Utilized appropriate data structures and algorithms to optimize performance where necessary, ensuring efficient execution.
- Added comprehensive inline comments to elucidate intricate sections of code and provide context for future maintainers.
- Applied defensive programming techniques, including error handling and input validation, to fortify the code against potential failures and vulnerabilities.
- Conducted thorough testing to validate the functionality and correctness of refactored code, ensuring that no regressions were introduced.

Rationale:
- Enhancing code quality and maintainability reduces technical debt and facilitates future development efforts.
- Refactoring improves code comprehension, making it easier for developers to understand, maintain, and extend the codebase over time.
- Addressing code quality issues proactively fosters a more robust and reliable software system, enhancing overall product quality and user satisfaction.

Scope/Category:
Code Quality Improvement
Refactoring

References:
"Clean Code: A Handbook of Agile Software Craftsmanship" by Robert C. Martin
"Refactoring: Improving the Design of Existing Code" by Martin Fowler

qinxutan committed:
Merge pull request #1 from qinxutan/A-Assertions

A-Assertions PR

qinxutan committed:
Merge pull request #2 from qinxutan/A-CodeQuality

Code Quality PR

qinxutan committed:
Set up of CI

qinxutan committed:
Detect Duplicates

qinxutan committed:
Checkstyle

qinxutan committed:
Checkstyle

qinxutan committed:
Checkstyle

ShanylOng commented:
Very clean and well documented code. The name of your chatbot is also very interesting! Keep up the great work.

qinxutan committed:
Better GUI

qinxutan committed:
Better GUI

qinxutan committed:
Give unique personality

qinxutan committed:
Error Handling

qinxutan committed:
Syntax edits

qinxutan committed:
Set main class

qinxutan committed:
User Guide edit

qinxutan committed:
Update build.gradle

qinxutan committed:
Syntax errors

qinxutan committed:
Syntax changes

qinxutan committed:
Remove additional main class
